lxd/networks_utils.go

```diff
@@ -1026,6 +1029,104 @@ func networkUpdateStatic(s *state.State, networkName string) error {
 	return nil
 }
 
+// networkUpdateForkdnsServersFile takes a list of node addresses and writes them atomically to
+// the forkdns.servers file ready for forkdns to notice and re-apply its config.
+func networkUpdateForkdnsServersFile(networkName string, addresses []string) error {
+	// We don't want to race with ourselves here
+	forkdnsServersLock.Lock()
+	defer forkdnsServersLock.Unlock()
+
+	permName := shared.VarPath("networks", networkName, "forkdns.servers/servers.conf")
+	tmpName := permName + ".tmp"
+
+	// Open tmp file and truncate
+	tmpFile, err := os.Create(tmpName)
+	if err != nil {
+		return err
+	}
+	defer tmpFile.Close()
+
+	for _, address := range addresses {
+		_, err := tmpFile.WriteString(address + "\n")
+		if err != nil {
+			return err
+		}
+	}
+
+	tmpFile.Close()
+
+	// Atomically rename finished file into permanent location so forkdns can pick it up.
+	err = os.Rename(tmpName, permName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// networkUpdateForkdnsServersTask runs every 30s and refreshes the forkdns servers list.
+func networkUpdateForkdnsServersTask(d *Daemon) (task.Func, task.Schedule) {
+	f := func(ctx context.Context) {
+		s := d.State()
+
+		// Get a list of managed networks
+		networks, err := s.Cluster.NetworksNotPending()
+		if err != nil {
+			logger.Errorf("Failed to get networks: %v", err)
+			return
+		}
+
+		for _, name := range networks {
+			n, err := networkLoadByName(s, name)
+			if err != nil {
+				logger.Errorf("Failed to load network: %v", err)
+				return
+			}
+
+			if n.config["bridge.mode"] == "fan" {
+				n.refreshForkdnsServerAddresses()
```

Comment:
I'm not seeing this in previous commits, are the commits mis-ordered?

Reply:
@stgraber yes I went through a few iterations on this and things have got a bit muddled, will rework the commits.